mesa_frames/abstract/space.py

```diff
@@ -321,3 +383,345 @@ def random(self) -> Generator:
         Generator
         """
         return self.model.random
+
+
+class DiscreteSpaceDF(SpaceDF):
+    """The DiscreteSpaceDF class is an abstract class that defines the interface for all discrete space classes (Grids and Networks) in mesa_frames.
+
+    Methods
+    -------
+    __init__(model: 'ModelDF', capacity: int | None = None)
+        Create a new DiscreteSpaceDF object.
+    is_free(pos: DiscreteCoordinate | DiscreteCoordinates) -> DataFrame
+        Check whether the input positions are free (there exists at least one remaining spot in the cells).
+    is_empty(pos: DiscreteCoordinate | DiscreteCoordinates) -> DataFrame
+        Check whether the input positions are empty (there isn't any single agent in the cells).
+    is_full(pos: DiscreteCoordinate | DiscreteCoordinates) -> DataFrame
+        Check whether the input positions are full (there isn't any spot available in the cells).
+    move_to_empty(agents: IdsLike | AgentContainer | Collection[AgentContainer], inplace: bool = True) -> Self
+        Move agents to empty cells in the space (cells where there isn't any single agent).
+    move_to_free(agents: IdsLike | AgentContainer | Collection[AgentContainer], inplace: bool = True) -> Self
+        Move agents to free cells in the space (cells where there is at least one spot available).
+    sample_cells(n: int, cell_type: Literal["any", "empty", "free", "full"] = "any", with_replacement: bool = True) -> DataFrame
+        Sample cells from the grid according to the specified cell_type.
+    get_neighborhood(radius: int | float | Sequence[int] | Sequence[float], pos: DiscreteCoordinate | Discrete
+        Get the neighborhood cells from a given position.
+    get_cells(cells: DiscreteCoordinates | None = None) -> DataFrame
+        Retrieve a dataframe of specified cells with their properties and agents.
+    set_cells(properties: DataFrame, cells: DiscreteCoordinates | None = None, inplace: bool = True) -> Self
+        Set the properties of the specified cells.
```

Comment:
Maybe add that this is to support capability of core Mesa's property layer?

Reply:
*similar capability. In core Mesa, you have to update the entire grid's property at once. But it seems here that we can do some filtering before hand because of DF. DF is more expressive than NumPy ndarray.

Reply:
I will update the docs to specify it.